pydantic/_internal/_core_utils.py

```diff
@@ -228,7 +247,7 @@ def handle_definitions_schema(self, schema: core_schema.DefinitionsSchema, f: Wa
             # This means we'd be returning a "trivial" definitions schema that just wrapped the inner schema
             return new_inner_schema
 
-        new_schema = schema.copy()
+        new_schema = schema
```

Comment:
Looks like similar question / concern about removal of copying here. I understand it's probably great for perf so just need to understand implications.

I would be somewhat glad if these could be split into a separate PR (followup?) given they are an orthogonal problem to the schema ref inlining changes.

Reply:
Happy to split this PR up